cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/sql/GcpDatabaseServerUpgradeService.java

```diff
@@ -1,21 +1,96 @@
+    private GcpSQLAdminFactory gcpSQLAdminFactory;
+
+    @Inject
+    private GcpStackUtil gcpStackUtil;
+
     @Override
     public List<CloudResource> upgrade(AuthenticatedContext ac, DatabaseStack stack, PersistenceNotifier resourceNotifier, TargetMajorVersion databaseVersion)
             throws Exception {
+        GcpDatabaseServerView databaseServerView = new GcpDatabaseServerView(stack.getDatabaseServer());
+        String deploymentName = databaseServerView.getDbServerName();
+        SQLAdmin sqlAdmin = gcpSQLAdminFactory.buildSQLAdmin(ac.getCloudCredential(), ac.getCloudCredential().getName());
+        String projectId = gcpStackUtil.getProjectId(ac.getCloudCredential());
+        String availabilityZone = ac.getCloudContext().getLocation().getAvailabilityZone().value();
+        List<CloudResource> buildableResource = new ArrayList<>();
+        buildableResource.add(getGcpDatabase(deploymentName, availabilityZone));
+
+        try {
+            InstancesListResponse list = sqlAdmin.instances().list(projectId).execute();
+            Optional<DatabaseInstance> first = Optional.empty();
+            if (!list.isEmpty()) {
+                first = list.getItems()
+                        .stream()
+                        .filter(e -> e.getName().equals(deploymentName))
+                        .findFirst();
+            }
+            if (!first.isEmpty()) {
+                DatabaseInstance databaseInstance = first.get();
+                databaseInstance.setDatabaseVersion(databaseServerView.getDatabaseType() + "_" + databaseVersion.getVersion());
+                SQLAdmin.Instances.Patch patch = sqlAdmin.instances().patch(projectId, databaseInstance.getName(), databaseInstance);
+                patch.setPrettyPrint(Boolean.TRUE);
+                try {
+                    Operation operation = patch.execute();
+                    verifyOperation(operation, buildableResource);
+                    CloudResource operationAwareCloudResource = createOperationAwareCloudResource(buildableResource.get(0), operation);
+                    databasePollerService.upgradeDatabasePoller(ac, List.of(operationAwareCloudResource));
+                    buildableResource.forEach(dbr -> resourceNotifier.notifyUpdate(dbr, ac.getCloudContext()));
+                    return Collections.singletonList(operationAwareCloudResource);
+                } catch (GoogleJsonResponseException e) {
+                    throw new GcpResourceException(checkException(e), resourceType(), buildableResource.get(0).getName());
+                }
+            } else {
+                LOGGER.debug("Deployment does not exist: {}", deploymentName);
+            }
+        } catch (GoogleJsonResponseException e) {
+            throw new GcpResourceException(checkException(e), resourceType(), buildableResource.get(0).getName());
+        }
         return List.of();
     }
+
+    public CloudResource getGcpDatabase(String deploymentName, String availabilityZone) {
```

Comment:
question: do we need to create a new `GCP_DATABASE` resource here? What happens with the original one? Cannot it cause problems that 2 exist or do we delete the other one?

Reply:
It won't be a new resource, the original one will be updated via the resourceNotifier.